registry_internal_test.go

```diff
@@ -0,0 +1,189 @@
+package taskrunner
+
+import (
+	"context"
+	"testing"
+
+	"github.com/samsarahq/taskrunner/shell"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddFlagsValidation(t *testing.T) {
+	mockRunWithFlags := func(ctx context.Context, shellRun shell.ShellRun, flags map[string]FlagArg) error {
+		return nil
+	}
+
+	testCases := []struct {
+		description string
+		shouldPanic bool
+		mockTask    *Task
+		panicVal    string
+	}{
+		{
+			description: "Should panic if a flag is missing a description",
+			mockTask: &Task{
+				Name:         "mock/task/1",
+				RunWithFlags: mockRunWithFlags,
+				Flags: []TaskFlag{
+					{
+						LongName:  "flagA",
+						ValueType: BoolTypeFlag,
+					},
+				},
+			},
+			panicVal: "Please provide a description for the `flagA` flag",
+		},
+		{
+			description: "Should panic if a flag is missing a ValueType",
+			mockTask: &Task{
+				Name:         "mock/task/1",
+				RunWithFlags: mockRunWithFlags,
+				Flags: []TaskFlag{
+					{
+						Description: "This is a description of flagA",
+						LongName:    "flagA",
+					},
+				},
+			},
+			panicVal: "Please set the flag ValueType with `taskrunner.StringTypeFlag`, `taskrunner.BoolTypeFlag` or `taskrunner.IntTypeFlag`.",
+		},
+		{
+			description: "Should panic if a flag is has ValueType that does not match XTypeFlag",
+			mockTask: &Task{
+				Name:         "mock/task/1",
+				RunWithFlags: mockRunWithFlags,
+				Flags: []TaskFlag{
+					{
+						Description: "This is a description of flagA",
+						LongName:    "flagA",
+						ValueType:   "thisIsInvalid",
```

Comment:
fwiw i guess there could also be floats and in the go std lib flag package, there is also a duration.

Would we want to support custom types, or just add those types?

Reply:
Hmm let's just add those in for now

Maybe we can support custom types in an iterative update?

Reply:
Added support for float64 [here](https://github.com/samsarahq/taskrunner/pull/48/commits/3f6e2185ce80a6d15ad0fb9ae5dd7cfc56dc2eea) and support for Duration [here](https://github.com/samsarahq/taskrunner/pull/48/commits/8679962ec8550fc5569eff3c2900ba6efe4cfd75)